packages/eslint-plugin/docs/rules/require-await.mdx

```diff
@@ -12,14 +12,45 @@ import TabItem from '@theme/TabItem';
 This rule extends the base [`eslint/require-await`](https://eslint.org/docs/rules/require-await) rule.
 It uses type information to allow promise-returning functions to be marked as `async` without containing an `await` expression.
 
+:::note
+Unlike the base rule, our implementation also supports [async generator functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function*).
+:::
+
 ## Examples
 
-Examples of **correct** code for this rule:
+<Tabs>
+<TabItem value="❌ Incorrect">
+
+```ts
+async function returnNumber() {
+  return 1;
+}
+
+async function* asyncGenerator() {
+  const num = Promise.resolve(1);
+  yield 1 + num;
```

Comment:
One more complaint (sorry!) - we try to avoid putting type errors in our examples that might be seen as red herrings. Here, TS will complain that `number` and `Promise<number>` can't be added to each other, which isn't relevant to what this rule is about. Let's fix that up so we don't confuse readers 🙂 

i.e. something like this?
```ts
// incorrect
async function* asyncGenerator() {
   yield 1;
}

// correct
function* asyncGenerator() {
   yield 1;
}
```